FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -125,18 +130,29 @@ final class MessageService {
                 isEmail: true
             )
 
-            let processedMessage = self.processMessage(rawMimeData: rawMimeData, with: decrypted)
+            let processedMessage = try self.processMessage(rawMimeData: rawMimeData, with: decrypted, keys: keys)
             resolve(processedMessage)
         }
     }
 
-    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg) -> ProcessedMessage {
+    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg, keys: [PrvKeyInfo]) throws -> ProcessedMessage {
         let decryptErrBlocks = decrypted.blocks
             .filter { $0.decryptErr != nil }
 
-        let attachments = decrypted.blocks
+        let attachments = try decrypted.blocks
             .filter(\.isAttachmentBlock)
-            .map(MessageAttachment.init)
+            .map { block -> MessageAttachment in
+                var name = block.attMeta?.name ?? "Attachment"
+                let size = block.attMeta?.length ?? 0
+                var data = block.attMeta?.data ?? Data()
```

Comment:
These three cannot should not be empty. If attMeta is missing, then it's not an attachment, and can be skipped. You can add a guard for that.

Reply:
just copy pasted init of MessageAttachment